recipesearch/src/Recipes/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,40 @@
+import React, { Component } from 'react'
+import './SearchBar.css';
+import  {searchRecipesAction} from './action';
+import { connect } from 'react-redux';
+
+export class SearchBar extends Component {
+    state = {
+        searchValue: '',
+    }
+
+    handleSearchChange = (event) => {
+        this.setState({
+            searchValue: event.target.value,
+        })
+    }
+
+    handleRecipeSearch = (event) =>{
+        const { searchValue } = this.state
+        event.preventDefault();
+        this.props.searchRecipesAction(searchValue);
+    }
+
+    render(){
+        return(
+      
+            <div>
+                <input type='text' className='searchInput' onChange={this.handleSearchChange}/>
+                <input type='submit' className='searchButton' onClick={this.handleRecipeSearch}/>
+            </div>
+        )
+    }
+}
+
+const mapDispatchToProps = dispatch =>{
+    return {
+        searchRecipesAction: (searchValue) => dispatch(searchRecipesAction(searchValue))
+    }
+}
+
+export const searchBarContainer = connect(null, mapDispatchToProps)(SearchBar)
```

Comment:
What happens if you don't pass in anything instead of passing in `null`

Reply:
I didnot test that out but since I wasnot passing mapstatetoprops that is why I put null.
Let me try it out also read about it so as to know what to do.

Reply:
@edmondatto  I have tested it out and when I remove null as the first argument am given an error of dispatch not being a function.
I have  read this documentation about [connect HOC](https://github.com/reduxjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options) and it highlights that when we are not passing mapStateToProps in the connect HOC you have to always pass null or undefined to indicate that we are not subscribing to any updates to the state.
The reverse is true because we can omit mapDisptachToProps from the connect HOC because there is a default mapStateToProps that injects dipatch method to outr component.